File: patterns/structural_fasade/structural_fasade_webref.js
class Сonveyor {

	setBody() {
		console.log('Body set!');
	}

	getEngine() {
        console.log('Dismantle Engine!');
    }

	setEngine() {
		console.log('Engine set!');
    }
    
    getInterior() {
		console.log('Dismantle Exterior!');
	}

	setInterior() {
		console.log('Exterior added!');
	}

	changeInterior() {
		console.log('Update interior!');
	}

	setExterior() {
		console.log('Added interior!');
	}

	setWheels() {
		console.log('Wheels!');
	}

	addElectronics() {
		console.log('Added electronics!');
	}

	paint() {
		console.log('Car painted!');
	}
}

class ConveyorFacade {
	constructor(car) {
		this.car = car;
	}

	assembleCar() {
		this.car.setBody();
		this.car.setEngine();
		this.car.setInterior();
		this.car.setExterior();
		this.car.setWheels();
		this.car.addElectronics();
		this.car.paint();
    }
    
    // -- 2
    changeEngine() {
        this.car.getEngine();
        this.car.setEngine();
    }

    changeInterior() {
        this.car.getInterior();
        this.car.setInterior();
    }
}

// ----
const conveyor = new ConveyorFacade(new Сonveyor());

let car = conveyor.assembleCar();

console.log(car);

// -- 2
car = conveyor.changeEngine();
car = conveyor.changeInterior();
console.log(car);

File: patterns/structural_fasade/structural_fasade_webref.py
class Conveyor:
  def set_body(self):
    print('body set!')

  def get_engine(self):
    print('Dismantle Engine!')

  def set_engine(self):
    print('engine set!')

class ConveyorFasade:
  def __init__(self, car):
    self.car = car

  def assemble_car(self):
    self.car.set_body()
    self.car.set_engine()

  def change_engine(self):
    self.car.get_engine()
    self.car.set_engine()
        

if __name__ == "__main__":
  # conveyor = Conveyor()
  conveyorFasade = ConveyorFasade(Conveyor())

  car = conveyorFasade.assemble_car()
  car = conveyorFasade.change_engine()
